test/node-rpc-test.js

```diff
@@ -5,43 +5,145 @@
+const {NodeClient, WalletClient} = require('bclient');
+const MerkleBlock = require('../lib/primitives/merkleblock');
+const {forValue} = require('./util/common');
+
+// main timeout for the tests.
+const TIMEOUT = 5000;
 
 const ports = {
-  p2p: 49331,
-  node: 49332,
-  wallet: 49333
+  full: {
+    p2p: 49331,
+    node: 49332,
+    wallet: 49333
+  },
+  spv: {
+    p2p: 49431,
+    node: 49432,
+    wallet: 49433
+  },
+  pruned: {
+    p2p: 49531,
+    node: 49532,
+    wallet: 49533
+  },
+  fullindexed: {
+    p2p: 49631,
+    node: 49632,
+    wallet: 49633
+  }
 };
 
-const node = new FullNode({
+const nodeOptions = {
   network: 'regtest',
   apiKey: 'foo',
   walletAuth: true,
   memory: true,
   workers: true,
   workersSize: 2,
-  plugins: [require('../lib/wallet/plugin')],
-  port: ports.p2p,
-  httpPort: ports.node,
-  env: {
-    'BCOIN_WALLET_HTTP_PORT': ports.wallet.toString()
-  }});
-
-const nclient = new NodeClient({
-  port: ports.node,
-  apiKey: 'foo',
-  timeout: 15000
-});
+  publicHost: '127.0.0.1',
+  plugins: [require('../lib/wallet/plugin')]
+};
 
 describe('RPC', function() {
-  this.timeout(15000);
+  this.timeout(TIMEOUT);
+
+  const getClientOpts = (port) => {
+    return {
+      apiKey: nodeOptions.apiKey,
+      timeout: TIMEOUT,
+      port: port
+    };
+  };
+
+  const fullnodeAddr = `127.0.0.1:${ports.full.p2p}`;
+  const fullnode = new FullNode({
+    listen: true,
+    bip37: true,
+    ...nodeOptions,
+    ...getNodePorts(ports.full)
+  });
+
+  const spvnode = new SPVNode({
+    only: fullnodeAddr,
+    ...nodeOptions,
+    ...getNodePorts(ports.spv)
+  });
+
+  const prunednode = new FullNode({
+    prune: true,
+    only: fullnodeAddr,
+    ...nodeOptions,
+    ...getNodePorts(ports.pruned)
+  });
+  const fullindexed = new FullNode({
+    only: fullnodeAddr,
+    indexTX: true,
+    ...nodeOptions,
+    ...getNodePorts(ports.fullindexed)
+  });
+
+  const nclient = new NodeClient({...getClientOpts(ports.full.node)});
+  const wclient = new WalletClient({...getClientOpts(ports.full.wallet)});
+
+  const spvnclient = new NodeClient({...getClientOpts(ports.spv.node)});
+  const prunednclient = new NodeClient({...getClientOpts(ports.pruned.node)});
+  const indexednclient = new NodeClient({
+    ...getClientOpts(ports.fullindexed.node)
+  });
+
+  const MINER_WALLET = 'test';
 
   before(async () => {
-    await node.open();
+    await fullnode.open();
+    await fullnode.connect();
+
+    await prunednode.open();
+    await prunednode.connect();
+    prunednode.startSync();
+
+    await spvnode.open();
+    await spvnode.connect();
+    spvnode.startSync();
+
+    await fullindexed.open();
+    await fullindexed.connect();
+    fullindexed.startSync();
+
+    // create and select miner wallet.
+    await wclient.createWallet(MINER_WALLET);
+    await wclient.execute('selectwallet', [MINER_WALLET]);
+
+    // setup miner wallet
+    const address = await wclient.execute('getnewaddress');
```

Comment:
For miner we could have `MinerContext` which will manage funds specifically to miner that will only act as faucet.